opm/test_util/summaryIntegrationTest.cpp

```diff
@@ -0,0 +1,352 @@
+	    if(volume.error == 0){
+		std::cout << "For keyword " << keyword << " the total production volume is 0" << std::endl; 
+	    }
+	    else{
+		std::cout << "For keyword " << keyword << " the total production volume is "<< volume.total << ", the error volume is " << volume.error << " the error ratio is " << volume.error/volume.total << std::endl;
+	    }
+	}  
+  	checkForKeyword(timeVec1, timeVec2, keyword);
+	return;
+    } 
+    OPM_THROW(std::invalid_argument, "The keyword used is not common for the two files.");
+  
+}
+
+
+void IntegrationTest::checkForKeyword(std::vector<double> timeVec1, std::vector<double> timeVec2, const char* keyword){
+    std::vector<double> dataVec1, dataVec2;
+  
+    getDataVecs(dataVec1,dataVec2,keyword); 
+    chooseReference(timeVec1, timeVec2,dataVec1,dataVec2);
+ 
+    if(allowSpikes){
+	checkWithSpikes(keyword);
+    }  
+
+    if(findVolumeError ||oneOfTheMainVariables ){
+	volumeErrorCheck(keyword);
+    }
+}
+
+
+
+
+
+int IntegrationTest::checkDeviation(Deviation deviation){
+    double absTol = getAbsTolerance();
+    double relTol = getRelTolerance();
+
+    if (deviation.rel> relTol && deviation.abs > absTol){
+	return 1;
+    }
+    return 0;
+}
+
+
+void IntegrationTest::findGreatestErrorRatio(WellProductionVolume volume, double &greatestRatio, const char* currentKeyword, std::string &keywordWithGreatestErrorRatio){
+    if (volume.total != 0 && volume.total - volume.error > getAbsTolerance()){
+	if(volume.error/volume.total > greatestRatio){
+	    greatestRatio = volume.error/volume.total; 
+	    std::string currentKeywordStr(currentKeyword);
+	    keywordWithGreatestErrorRatio = currentKeywordStr; 
+	}
+    }
+}
+
+
+void IntegrationTest::volumeErrorCheck(const char* keyword){
+    std::string keywordString(keyword);
+    std::string firstFour = keywordString.substr(0,4);
+    std::string fifth = keywordString.substr(4,1); //To make sure we do not include history vectors.
+   
+    if (mainVariable.empty()){
+	updateVolumeError(keyword);
+    }
+    
+    if(firstFour ==  mainVariable && fifth!="H"){
+	if(mainVariable == "WOPR"){
+	    WellProductionVolume result =  getWellProductionVolume(keyword);
+	    WOPR.total += result.total; 
+	    WOPR.error += result.error; 
+	}
+	if(mainVariable == "WWPR"){
+	    WellProductionVolume result =  getWellProductionVolume(keyword);
+	    WWPR.total += result.total; 
+	    WWPR.error += result.error; 
+	}
+
+	if(mainVariable == "WGPR"){
+	    WellProductionVolume result =  getWellProductionVolume(keyword);
+	    WGPR.total += result.total; 
+	    WGPR.error += result.error; 
+	}
+
+	if(mainVariable == "WBHP"){
+	    WellProductionVolume result =  getWellProductionVolume(keyword);
+	    WBHP.total += result.total; 
+	    WBHP.error += result.error; 
+	}	
+    }
+}
+
+
+void IntegrationTest::updateVolumeError(const char* keyword){
+    std::string keywordString(keyword);
```

Comment:
1. I'm not sure how to use the smspec_node_type to get the information needed. 

2. I do agree that the best solution for finding e.g. the total production of a well is to use the last data value of the total keywords. This is what I do in the `volumeUpdateError(keyword)` function. The reason for integrating up the rate to a total is that it is the easiest way to check whether the error rate (total error/total production) is acceptable. I'm not saying it is the best way, however it is quite simple. An advantage is that it will not be possible to compare the total error from the well of interest with the total production from another well.  

3.  > Integrating WBHP does not give a volume

True. I have commented on this in the `summaryIntegrationTest.hpp`. For simplicity, I will treat the integrated WBHP as a volume.

I'm considering removing the WellProductionVolume member variables of the class, and rather using double values, e.g. WOPT and WOPError.